src/main/java/no/difi/asic/AsicUtils.java

```diff
@@ -120,21 +124,20 @@ public static void combine(OutputStream outputStream, InputStream... inputStream
     }
 
     public static MimeType detectMime(String filename) throws IOException {
-        // Use Files to find content type
-        String mimeType = Files.probeContentType(Paths.get(filename));
-
-        // Use URLConnection to find content type
-        if (mimeType == null) {
-            logger.info("Unable to determine MIME type using Files.probeContentType(), trying URLConnection.getFileNameMap()");
-            mimeType = URLConnection.getFileNameMap().getContentTypeFor(filename);
+        String mimeType = null;
+        final Path path = Paths.get(filename);
+        if (Files.exists(path)) {
+            mimeType = TIKA.detect(path);
         }
 
-        // Throw exception if content type is not detected
         if (mimeType == null) {
-            throw new IllegalStateException(String.format("Unable to determine MIME type of %s", filename));
+            logger.info("Unable to determine MIME type using tika.detect(Path), trying tika.detect(String)");
+            mimeType = TIKA.detect(filename);
+            if (mimeType == null || TIKA_FALLBACK_MIME.equals(mimeType)) {
```

Comment:
Det er ikkje ønskelig å feile sjølv om content type er `application/octet-stream`. Det kan være eit gyldig case for andre brukere av biblioteket.